src/coreclr/jit/emitxarch.cpp

```diff
@@ -2651,22 +2658,64 @@ emitter::instrDesc* emitter::emitNewInstrAmdCns(emitAttr size, ssize_t dsp, int
     }
 }
 
-/*****************************************************************************
- *
- *  The next instruction will be a loop head entry point
- *  So insert a dummy instruction here to ensure that
- *  the x86 I-cache alignment rule is followed.
- */
-
-void emitter::emitLoopAlign()
+//-----------------------------------------------------------------------------
+//
+//  The next instruction will be a loop head entry point
+//  So insert a dummy instruction here to ensure that
+//  the x86 I-cache alignment rule is followed.
+//
+void emitter::emitLoopAlign(unsigned short paddingBytes)
 {
     /* Insert a pseudo-instruction to ensure that we align
        the next instruction properly */
 
-    instrDesc* id = emitNewInstrSmall(EA_1BYTE);
-    id->idIns(INS_align);
-    id->idCodeSize(15); // We may need to skip up to 15 bytes of code
-    emitCurIGsize += 15;
+    paddingBytes       = min(paddingBytes, 15); // We may need to skip up to 15 bytes of code
```

Comment:
Should we assert that the value is <= 15?

Reply:
Also is there some symbolic constant for the `15`?

Reply:
Added `#define MAX_CODE_SIZE 15` for xarch. let me know if you have better name suggestion.